src/coreComponents/mainInterface/ProblemManager.cpp

```diff
@@ -491,7 +491,7 @@ void ProblemManager::generateMesh()
 
   MeshManager & meshManager = this->getGroup< MeshManager >( groupKeys.meshManager );
   meshManager.generateMeshes( domain );
-  Group & cellBlockManager = domain.getGroup( keys::cellManager );
+  CellBlockManagerABC & cellBlockManager = domain.getGroup< CellBlockManagerABC >( keys::cellManager );
```

Comment:
I think this could be interesting indeed.
To be noted that this induces a change of dependencies, since now `DomainPartition` does not depend on `CellBlockManager`. I'll have a look.
Also, during the review/merge phase, I wanted to reduce the footprint of the PR to minimize conflicts.